### DISCUSSION TITLE
MantineHub – Modern, Pre-Configured Themes , UI blocks, and much more.

### DISCUSSION THREAD

**Discussion by Abhishek-S-Lal** [mantinedev/mantine#7056] [Show]
Hello Mantine Community! 👋

I’m excited to share **Mantine Theme Builder — a tool designed to make theming faster and easier** for Mantine users. Inspired by modern designs like Shadcn, this tool provides pre-configured themes that can be instantly applied to your project. It’s perfect for those who want a polished look without spending hours on custom styles.

### Project Overview
When I started with Mantine a couple of months ago, I loved its flexibility and extensive components, but I found myself wishing for a quicker way to achieve a modern, cohesive aesthetic. Mantine Theme Builder bridges that gap by offering ready-made themes that are fully customizable.

### How to Use
Choose a Theme: Browse pre-configured themes on the website.
Copy the Theme Config: Pick a style that fits your project, then copy the configuration.
Paste & Customize: Add it to your Mantine theme provider setup. Tweak it as needed to match your unique look.

### 🚀 Recent Updates
Mantine Theme Builder has evolved significantly since its initial launch:

1. ✨ Rebranded as MantineHub: We’ve expanded the scope of the project! MantineHub now includes themes and a real-time Playground to help you customize and visualize your themes and pre-built UI components to quickly develop your application.
2. 🎨 Complete Theming: All 100+ Mantine components are now fully themed, inspired by modern styles like Shadcn and Radix.
3. ⛹️‍♂️ Preview Playground: Check out how each component looks across themes and customize props like variants, colors, and more in real-time.
4. 🧩 Introducing Blocks: Blocks is a new experimental feature! We’ve included a small set of pre-built components and added a button to request the blocks you need. If this feature sparks enough interest, we’d love to bring in many more blocks based on your feedback. Let us know what you think about this feature!

### Plans & Contributions
The project is evolving, and I’d love to get feedback and contributions from the community! Planned features include adding more theme styles, ensuring compatibility across all Mantine components, and offering guides on customization.

I’m still fairly new to Mantine, so if you have suggestions for improving the theming approach or if there are features you’d like to see, please let me know. Contributions are welcome to help make this tool even more valuable for the community!

🌐 Website: [https://mantinehub.com](https://mantinehub.com)
📂 GitHub: [https://github.com/RubixCube-Innovations/mantine-theme-builder](https://github.com/RubixCube-Innovations/mantine-theme-builder)

Looking forward to your feedback, and happy theming! 😊

**Comment by AndrejNemec** [comment]
Thank you for creating such an incredible theme builder for Mantine – it’s a game-changer for my workflow!

**Comment by Sajarin-M** [comment]
Nice work! 👍

**Comment by icflorescu** [comment]
Fantastic work!
I'd say this is one of the most valuable Mantine-related projects that have appeared in the community lately.

**Reply by Shadowfita** [reply]
Completely agree! Pairs well with a certain context menu and datatable project. 😉

**Comment by pc-gs** [comment]
Amazing project, very useful. congrats @Abhishek-S-Lal 👏🏻

**Comment by hjhjdev** [comment]
Nice work!

**Comment by bhaireshm** [comment]
Does this support Nested MantineProvider?

[https://github.com/orgs/mantinedev/discussions/4803](https://github.com/orgs/mantinedev/discussions/4803)

**Comment by Shadowfita** [comment]
I completely share the same sentiment as @icflorescu. Ever since discovering mantine I've felt that this sort of complimentary tool was severely lacking, and would elevate the use of mantine, at least for me, significantly. I'm very excited to see that something is finally here, and of this quality.

**Comment by furkancakici** [comment]
Amazing work, thank you @Abhishek-S-Lal

**Comment by gotzmann** [comment]
Fantastic! Would love to see it as a full blown Mantine UI library of hundreds blocks for web and marketing

**Comment by nepton-mattpike** [comment]
Awesome, thank you

**Comment by alexthebake** [comment]
I'd love to see the radix version!

**Comment by jorcelinojunior** [comment]
Hi @Abhishek-S-Lal, thanks for sharing! The live preview and ready-made themes are super useful. Great job! 👏👏
